src/chess/bitboard.h

```diff
@@ -190,17 +190,29 @@ class Move {
  public:
   enum class Promotion : std::uint8_t { None, Queen, Rook, Bishop, Knight };
   Move() = default;
-  Move(BoardSquare from, BoardSquare to) : from_(from), to_(to) {}
-  Move(BoardSquare from, BoardSquare to, Promotion promotion)
-      : from_(from), to_(to), promotion_(promotion) {}
+  Move(BoardSquare from, BoardSquare to) {
+    data_ = 0;
+    data_ |= to.as_int();
+    data_ |= from.as_int() << 6;
+  }
+  Move(BoardSquare from, BoardSquare to, Promotion promotion){
+    data_ = 0;
+    data_ |= to.as_int();
+    data_ |= from.as_int() << 6;
+    data_ |= (uint8_t)promotion << 12;
+  }
   Move(const std::string& str, bool black = false);
   Move(const char* str, bool black = false) : Move(std::string(str), black) {}
 
-  BoardSquare from() const { return from_; }
-  BoardSquare to() const { return to_; }
-  Promotion promotion() const { return promotion_; }
-  bool IsCastling() const { return castling_; }
-  void SetCastling() { castling_ = true; }
+  BoardSquare to() const      { return (BoardSquare)((data_ & 0b0000000000111111)); }
```

Comment:
possibly would be good to define constants for this magic numbers?

Reply:
Yes, absolutely.

Something like this:

```
private:
  enum MASKS_ : uint16_t {
    TO_MASK_     = 0b0000000000111111;
    FROM_MASK_   = 0b0000111111000000;
    PROMO_MASK_  = 0b0111000000000000;
    CASTLE_MASK_ =  0b1000000000000000;
  }
```

Then you can do e.g. `void SetFrom(BoardSquare from) { data_ = (data_ & ~FROM_MASK_) | (from.as_int() << 6); }`

Reply:
Done